### PR TITLE
Update passwords and blacklist when deleting

### LINKED ISSUE #8906: Deleting saved password still shown until page refresh
## Test plan

https://github.com/brave/browser-laptop/pull/9085#issue-231739942

----

- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
Deleting saved password still shown until page refresh

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows 10 x64

- Brave Version (revision SHA):
Brave	0.15.304
rev	b872523
Muon	3.0.103

- Steps to reproduce:
    1. Keep Brave builtin password manager
    2. Visit any site and save the login details
    3. Open about:passwords, click on the delete button next to the saved entry, doesn't clear, reload the page and the entry is gone
    
- Actual result:
Deleting saved password still shown until page refresh

- Expected result:
Should clear the entry from table after clicking the delete button

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
No

- Can this issue be consistently reproduced?
Yes

- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:
passwords

- Any related issues:

### PR DESCRIPTION
**Pull request by darkdh**
fix #8906
resolve item 1 of #8874

Auditors: @diracdeltas, @bridiver

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:
1. Make sure you have passwords and never saved sites in
about:passwords
2. delete any of them or clear all passwords should relfect changes
instantly
Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [x] New files have MPL2 license header

### CODE REVIEW

**Comment by diracdeltas** [comment]
I tried this and it didn't seem to fix the updating issue. I deleted a 'never save passwords on this site' item in about:passwords and it remained there until i switched tabs.

**Comment by darkdh** [comment]
@diracdeltas , I think it works now with 9322a5f

File: app/autofill.js [hunk 1]
```diff
@@ -68,3 +69,15 @@ module.exports.removeLogin = (form) => {
 module.exports.clearLogins = (form) => {
   session.defaultSession.autofill.clearLogins(form)
 }
+
+module.exports.getAutofillableLogins = (tab) => {
+  session.defaultSession.autofill.getAutofillableLogins((result) => {
+    tab.send(messages.PASSWORD_DETAILS_UPDATED, result)
```
Comment (bridiver) [inline app/autofill.js]: don't use ipc directly, use actions instead

File: js/about/passwords.js [hunk 2]
```diff
@@ -64,6 +64,7 @@ class SiteItem extends React.Component {
 
   onDelete () {
     aboutActions.deletePassword(this.props.site.toJS())
+    ipc.send(messages.UPDATE_PASSWORD_SITE_DETAILS)
```
Comment (bridiver) [inline js/about/passwords.js]: nothing additional is needed here. I would suggest creating a passwordsReducer and handling `aboutAction.deletePassword` in that

File: js/about/passwords.js [hunk 3]
```diff
@@ -165,6 +167,8 @@ class AboutPasswords extends React.Component {
       'This cannot be undone.'
     if (window.confirm(msg)) {
       aboutActions.clearPasswords()
+      ipc.send(messages.UPDATE_PASSWORD_DETAILS)
```
Comment (bridiver) [inline js/about/passwords.js]: same as above, no addition messages are needed here

**Comment by darkdh** [comment]
@bridiver, updated in cd84fbb9cedbf6291909966fca29c88c5c443dab.
Please review again

File: app/browser/reducers/passwordManagerReducer.js [hunk 4]
```diff
@@ -230,6 +257,15 @@ const passwordManagerReducer = (state, action, immutableAction) => {
     case appConstants.APP_UPDATE_PASSWORD:
       updatePassword(action.get('username'), action.get('origin'), action.get('tabId'))
       break
+    case appConstants.APP_REMOVE_PASSWORD:
+      removePassword(action.get('passwordDetail').toJS(), action.get('senderTabId'))
```
Comment (bridiver) [inline app/browser/reducers/passwordManagerReducer.js]: ++ much better!

File: app/autofill.js (hunk 1 above)
Comment (bridiver) [inline app/autofill.js]: we should add a null/isDestroyed check here and the one below

File: app/browser/reducers/passwordManagerReducer.js [hunk 5]
```diff
@@ -176,6 +176,33 @@ const updatePassword = (username, origin, tabId) => {
   }
 }
 
+const removePassword = (form, tabId) => {
+  autofill.removeLogin(form)
+  const webContents = getWebContents(tabId)
+  setTimeout(() => {
```
Comment (bridiver) [inline app/browser/reducers/passwordManagerReducer.js]: why the 1 second timeout? Can we just do `setImmediate`?

Comment (darkdh) [inline app/browser/reducers/passwordManagerReducer.js]: we need to add password store observer on muon per discussion.

**Comment by darkdh** [comment]
@bridiver, 
fixed it with https://github.com/brave/muon/pull/209

**Comment by bridiver** [comment]
I just approved it, but want to make sure there is test coverage before merging

**Comment by darkdh** [comment]
recorded here
https://github.com/brave/browser-laptop/issues/9346

**Comment by bsclifton** [comment]
@darkdh don't forget to cherry-pick back into `0.18.x` and `0.17.x`

**Comment by darkdh** [comment]
will do! thanks for reminding

**Pull request merged**